### PR TITLE
added plugin that checks the status of websites

### PR DESCRIPTION
**Pull request by meskarune**
Ok, new plugin code on its own branch, I hope this is right XD

### CODE REVIEW

File: src/plugins/isup.lua [hunk 1]
```diff
@@ -0,0 +1,36 @@
+url = require("socket.url")
```
Comment (HalosGhost) [inline src/plugins/isup.lua]: Both of these should be `local` variables (don't want to pollute the global namespace unnecessarily).

File: src/plugins/isup.lua [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+plugin.help = 'Usage: isup <website>'
+
+plugin.main = function(args)
+    local website = args.message
+    if args.modules.config.debug then
+        print(website)
+    end
+    if not website then
+        args.modules.irc.privmsg(args.target, 'Give me a website or hostname to check')
```
Comment (HalosGhost) [inline src/plugins/isup.lua]: probably, we want to `return` at this point so we don't keep going through the rest (which would likely explode at line 23 (on `website:find…`)

File: src/plugins/isup.lua [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    end
+    local address = website:find('^https?://') and website or ('http://%s'):format(website)
```
Comment (HalosGhost) [inline src/plugins/isup.lua]: I like this simplification, but now seeing that `address` is only used on the next line, can we just inline it?

something like this:

```lua
local response, httpcode = https.request(('%s%s'):format(website:find('^https?://') and '' or 'http://', website))
```
Does that feel too compacted?

File: src/plugins/isup.lua [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    local response,httpcode = https.request(address)
+    if response then
+        local code = string.sub(tostring(httpcode), 1, 1)
+        local reply = messages[tonumber(code)]:format(httpcode)
+        args.modules.irc.privmsg(args.target, reply)
+        return
```
Comment (HalosGhost) [inline src/plugins/isup.lua]: these `return`s are unnecessary; there is no more text in the function following these statements.

File: src/plugins/isup.lua (hunk 3 above)
Comment (meskarune) [reply src/plugins/isup.lua]: This seems a bit harder to read, but maybe if its broken into more than one line?
```
local response, httpcode = https.request(('%s%s')
                           :format(website:find('^https?://') and '' or 'http://', website))
```

I'm not totally sure how to do multiple lines with lua

Comment (HalosGhost) [reply src/plugins/isup.lua]: eh, let's not worry about it for now.

**Pull request merged**